### PR TITLE
Chen And Garry's Ice Cream: Ice Cream DLC

### PR DESCRIPTION
**Pull request by YakumoChen**
## About The Pull Request
I scream for ice cream!!

Introduces many new flavours of ice cream:
-Caramel
-Banana
-Lemon Sorbet
-Orange Creamsicle
-Possibly more?? (I might add peach and some other ideas, but peach juice is hard to get)

image
(colours subject to change)

Slightly improved the ice cream vat code: 
-Chen And Garry's Original Flavours no longer use icon states and simply recolour the greyscale ice cream sprite like custom flavours already do
-It's easier to add more flavours as a result now

## Why It's Good For The Game
I always thought the ice cream vat could use more flavours. The custom flavour besides, it isn't as intuitive to rename the cone and the added variety is good. The lack of a banana flavour already was questionable. 
All the ice cream flavours used a selection of five sprites, now it's just one sprite and better supporting more additions.
Some of the flavours don't use milk! You can't do this with the custom flavour, making it slightly more interesting.

## Changelog
:cl:
add: Chen And Garry's Ice Cream is proud to debut a wide selection of cool new flavours on a space station near you!
code: Ice cream flavours now are all greyscaled similarly to GAGs
/:cl:

### CODE REVIEW

**Comment by YakumoChen** [comment]
uhh
hmmmmmmmm
I don't know how to solve this new test oroblem since I removed all the icon_states from ice cream...

**Comment by DAKKA-WAAAGH** [comment]
### **BASED**

**Comment by Jackal-boop** [comment]
The ice cream maker could use a TGUI overhaul

File: code/datums/components/food/ice_cream_holder.dm [hunk 1]
```diff
@@ -199,8 +199,8 @@ GLOBAL_LIST_INIT_TYPED(ice_cream_flavours, /datum/ice_cream_flavour, init_ice_cr
 /datum/ice_cream_flavour
 	/// Make sure the same name is not found on other types; These are singletons keyed by their name after all.
 	var/name = "Coderlicious Gourmet Double Deluxe Undefined"
-	/// The icon state of the flavour, overlay or not.
-	var/icon_state = "icecream_vanilla"
+	/// The colour of the ice cream for non-custom flavours.
+	var/color = "#F2EEDE"
```
Comment (Mothblocks) [inline code/datums/components/food/ice_cream_holder.dm:203]: Use the color defines

File: code/modules/food_and_drinks/machinery/icecream_vat.dm [hunk 2]
```diff
@@ -23,7 +23,12 @@
 		/datum/reagent/consumable/coco = 6,
 		/datum/reagent/consumable/vanilla = 6,
 		/datum/reagent/consumable/berryjuice = 6,
-		/datum/reagent/consumable/ethanol/singulo = 6)
+		/datum/reagent/consumable/ethanol/singulo = 6,
+		/datum/reagent/consumable/lemonjuice = 6,
+		/datum/reagent/consumable/caramel = 6,
+		/datum/reagent/consumable/banana = 6,
+		/datum/reagent/consumable/orangejuice = 6,
+		/datum/reagent/consumable/cream = 6)
```
Comment (Mothblocks) [inline code/modules/food_and_drinks/machinery/icecream_vat.dm:31]:
```suggestion
		/datum/reagent/consumable/cream = 6,
	)
```

File: code/datums/components/food/ice_cream_holder.dm (hunk 1 above)
Comment (YakumoChen) [reply code/datums/components/food/ice_cream_holder.dm:203]: Will look into that

**Comment by YakumoChen** [comment]
I need to figure out how to tackle the problem with robot customers and I'm not sure my funny code will work the same way (and I'll probably get help)
Or I could add the icon state var back and do nothing but that's code debt

**Comment by Helg2** [comment]
What about mint ice cream?

**Comment by Helg2** [comment]
> There isn't a reliable source of mint extract AFAIK 
Especially given mint extract can kill you

I mean yeah, but that's why it would be rare.

**Comment by Overuseddart** [comment]
> 

But it would taste REALLY good so it’s worth it

**Comment by YakumoChen** [comment]
Might need to redo this for merge conflicts but I've been on vacation so I haven't gotten to it yet, sorry

**Comment by YakumoChen** [comment]
Gonna get back to this soon but I've got new ideas I need to write down
After thinking about it, I can solve the lizard (perhaps moth too) problem by just making a lizard friendly cone to put ice cream in.

**Review by Twaticus — APPROVED** [review]

**Comment by optimumtact** [comment]
open a New pr when you're actually ready to finish this.

**Pull request closed without merging**